to.etc.domui/src/main/java/to/etc/domui/util/db/BatchUtil.java

```diff
@@ -0,0 +1,76 @@
+package to.etc.domui.util.db;
+
+import com.google.common.collect.Iterables;
+import org.eclipse.jdt.annotation.NonNull;
+import to.etc.domui.component.meta.ClassMetaModel;
+import to.etc.domui.component.meta.PropertyMetaModel;
+import to.etc.webapp.qsql.JdbcUtil;
+import to.etc.webapp.query.IIdentifyable;
+import to.etc.webapp.query.QField;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+final public class BatchUtil {
+
+	private BatchUtil() {}
+
+	/**
+	 * Does bulk delete using JDBC. Use it when performance matters ;)
+	 * @throws SQLException
+	 */
+	public static <K extends Number, T extends IIdentifyable<K>> boolean bulkDelete(@NonNull Connection con, ClassMetaModel cmm, @NonNull List<T> items) throws SQLException {
+		final PropertyMetaModel<?> pkPmm = cmm.getPrimaryKey();
+		if(null == pkPmm) {
+			throw new IllegalArgumentException("Unsupported working with class models that do not have PK columns! For " + cmm.getTableName() + " fund no PK!");
```

Comment:
Text is strange (2 times). Should be "Found no PK for " + tablename), and "Unsupported" should be "xxxxx is not supported" or "xxxxxx is unsupported".